app/assets/javascripts/discourse/mixins/upload.js.es6

```diff
@@ -74,6 +74,31 @@ export default Em.Mixin.create({
       }
     });
 
+    // all pre-processing or confirmations come here before file is submitted to server
+    // set autoUpload: false in uploadOptions to allow this function to be called
+    $upload.on("fileuploadadd", (e, data) => {
+      const options = this.uploadOptions();
+      if (options.shouldConfirm) {
```

Comment:
I'm not sure we need this boolean. Isn't setting `autoUpload` to `false` enough?

Reply:
My idea behind adding it was - let's say in future someone else wants to do some other processing or show something other than confirmation box, they can just add flags for it.

But I guess its ok for now since we don't have any other use-case. This can be added later.

Reply:
IHMO developing a generic solution is almost always a waste of time unless you already **know** about another use-case. It's highly likely that it's never needed in the future or that the generic solution doesn't work in some kind of unforseen use-case.